util/admin/admin.go

```diff
@@ -582,34 +606,13 @@ func rowWithCols(sessCtx sessionctx.Context, txn kv.Retriever, t table.Table, h
 			}
 			continue
 		}
-		// If have virtual generate column , decode all columns.
-		if col.IsGenerated() && col.GeneratedStored == false {
-			genColFlag = true
-		}
-		colTps[col.ID] = &col.FieldType
-	}
-	// if have virtual generate column, decode all columns
-	if genColFlag {
-		for _, c := range t.Cols() {
-			if c.State != model.StatePublic {
-				continue
-			}
-			colTps[c.ID] = &c.FieldType
-		}
 	}
 
-	rowMap, err := tablecodec.DecodeRow(value, colTps, sessCtx.GetSessionVars().Location())
+	rowMap, err := rowDecoder.DecodeAndEvalRowWithMap(sessCtx, value, sessCtx.GetSessionVars().Location(), time.UTC, nil)
```

Comment:
Why the time.Location change to time.UTC, not sessCtx.GetSessionVars().Location()?

Reply:
There is not change.
`sessCtx.GetSessionVars().Location()` is for `tablecodec.DecodeRow` to decode data.
`time.UTC` is for evaluated virtual generated column.